system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -0,0 +1,1128 @@
+
+                    // Validate the token, which also caches it if
+                    // validation passes
+                    ensureKeyLoaded();
+                    if (verifyToken(foundToken, userName.str(), foundRefreshToken, "", ""))
+                    {
+                        // Load parsed token out of cache
+                        gUserCache.get(userName.str(), userInfo);
+                    }
+                    else
+                    {
+                        return SecAccess_Unavailable;
+                    }
+                }
+                else
+                {
+                    // No token in the K/V store;
+                    return SecAccess_Unavailable;
+                }
+            }
+
+            if (key && stricmp(key,"Scope") == 0)
+                resultFlag = _authorizeEx(RT_FILE_SCOPE, *user, obj, nullptr, userInfo);
+            else if (key && stricmp(key,"workunit") == 0)
+                resultFlag = _authorizeEx(RT_WORKUNIT_SCOPE, *user, obj, nullptr, userInfo);
+        }
+
+        return resultFlag;
+    }
+
+    virtual bool checkScopeScans()
+    {
+        return true;
+    }
+
+    virtual unsigned getLDAPflags()
+    {
+        return DLF_ENABLED | DLF_SCOPESCANS; // | DLF_SAFE;
+    }
+
+    virtual void setLDAPflags(unsigned flags)
+    {
+        // Do nothinig
+    }
+
+    virtual bool clearPermissionsCache(IUserDescriptor *udesc)
+    {
+        return true;
+    }
+
+    virtual bool enableScopeScans(IUserDescriptor *udesc, bool enable, int *err)
+    {
+        return true;
+    }
+
+
+private:
+    mutable CriticalSection     crit;                       //!< Protects access to secretsPath contents
+    StringBuffer                clientID;                   //!< URL or unique name of the current cluster; from configuration
+    StringBuffer                loginEndpoint;              //!< Full URL to login endpoint; from configuration
+    StringBuffer                refreshEndpoint;            //!< Full URL to refresh endpoint; from configuration
+    bool                        allowSelfSignedCert;        //!< true = allow self-signed certificate; from configuration
+    StringBuffer                secretsPath;                //!< Secrets vault space or subdirectory name containing key; from configuration; @see ensureKeyLoaded()
+    time_t                      keyFilePathLastChecked;     //!< Timestamp of last time keyFilePath was checked for modification; @see ensureKeyLoaded()
+    SecAccessFlags              defaultPermAccess;          //!< Default permission value for permissions; from configuration
+    SecAccessFlags              defaultWUAccess;            //!< Default permission value for workunit scopes; from configuration
+    SecAccessFlags              defaultFileAccess;          //!< Default permission value for file scopes; from configuration
+    std::string                 keyContents;                //!< Contents of secret key; @see ensureKeyLoaded()
```

Comment:
not sure about mixed use of StringBuffer and std::string in same new class, would be clearer if consistent.